Client/Frontend/Fakespot/ShoppingProduct.swift

```diff
@@ -55,4 +73,26 @@ class ShoppingProduct: FeatureFlaggable {
 
         return Product(id: id, host: host, topLevelDomain: tld, sitename: sitename)
     }()
+
+    /// Fetches the analysis data for a specific product.
+    ///
+    /// - Returns: An instance of `ProductAnalysisData` containing the analysis data for the product, or `nil` if the product is not available.
+    /// - Throws: An error of type `Error` if there's an issue during the data fetching process.
+    /// - Note: This function is an asynchronous operation and should be called within an asynchronous context using `await`.
+    ///
+    func fetchProductAnalysisData() async throws -> ProductAnalysisData? {
```

Comment:
Do the unit tests need to be updated with this code change? Also @lmarceau would probably want a test for leaks added as well ;)

Reply:
Yeahhh leaks testing 😆 🚀